### ISSUE TITLE
Empty 'main.fmf' makes lint fail

### ISSUE THREAD

**Issue by lukaszachy** [teemtee/tmt#1331]
`tmt t ls` prints names as expected:
```
/a
```
However `tmt t lint` fails:
```
/a
pass test script must be defined
pass directory path must be absolute
pass directory path must exist
warn summary is very useful for quick inspection
Expected dictionary in yaml data, got 'NoneType'.
```

I don't think we forbid empty 'main.fmf' in specification so lint should cope with that.

To reproduce:
```
tmt init
echo 'test: ls' > main.fmf
mkdir a
touch a/main.fmf
```

**Comment by psss** [comment]
This seems to be related to recent typing changes, the `yaml_to_dict()` function is now a bit more strict. I guess, this should be fairly easy to fix, marking as such.

**Comment by idorax** [comment]
Yes, it's pretty easy to fix. If the `main.fmf` is empty, `loaded_data` is `None`, which should not be valiated via `if not isinstance(loaded_data, dict)`.

```python
(Pdb) s
--Call--
> /home/huanli/.virtualenvs/tmt/lib/python3.10/site-packages/tmt/utils.py(630)read()
-> def read(self, path: str, level: int = 2) -> str:
(Pdb) ll
630  ->	    def read(self, path: str, level: int = 2) -> str:
631  	        """ Read a file from the workdir """
632  	        if self.workdir:
633  	            path = os.path.join(self.workdir, path)
634  	        self.debug(f"Read file '{path}'.", level=level)
635  	        try:
636  	            with open(path, encoding='utf-8', errors='replace') as data:
637  	                return data.read()
638  	        except OSError as error:
639  	            raise FileError(f"Failed to read '{path}'.\n{error}")
(Pdb) b 637
Breakpoint 4 at /home/huanli/.virtualenvs/tmt/lib/python3.10/site-packages/tmt/utils.py:637
(Pdb) c
> /home/huanli/.virtualenvs/tmt/lib/python3.10/site-packages/tmt/utils.py(637)read()
-> return data.read()
(Pdb) pp data.read()
''
(Pdb) c
> /home/huanli/.virtualenvs/tmt/lib/python3.10/site-packages/tmt/utils.py(1198)yaml_to_dict()
-> yaml = YAML(typ=yaml_type)
(Pdb) ll
1195 	def yaml_to_dict(data: Any,
1196 	                 yaml_type: Optional[YamlTypType] = None) -> Dict[Any, Any]:
1197 	    """ Convert yaml into dictionary """
1198B->	    yaml = YAML(typ=yaml_type)
1199 	    loaded_data = yaml.load(data)
1200 	    if not isinstance(loaded_data, dict):
1201 	        raise GeneralError(
1202 	            f"Expected dictionary in yaml data, "
1203 	            f"got '{type(loaded_data).__name__}'.")
1204 	    return loaded_data
(Pdb) n
> /home/huanli/.virtualenvs/tmt/lib/python3.10/site-packages/tmt/utils.py(1199)yaml_to_dict()
-> loaded_data = yaml.load(data)
(Pdb) n
> /home/huanli/.virtualenvs/tmt/lib/python3.10/site-packages/tmt/utils.py(1200)yaml_to_dict()
-> if not isinstance(loaded_data, dict):
(Pdb) p loaded_data
None

#^^^ loaded_data is None !!!
```

**Comment by idorax** [comment]
Hi @lukaszachy and @psss, with the fix of PR#1332, the output of `tmt t lint` looks good. Please help to review, thanks!

* https://github.com/teemtee/tmt/pull/1332

```bash
(tmt) huanli@kvm-01-guest01:foo$ tree
.
├── a
│   └── main.fmf
└── main.fmf

1 directory, 2 files
(tmt) huanli@kvm-01-guest01:foo$ cat main.fmf 
test: ls
(tmt) huanli@kvm-01-guest01:foo$ cat a/main.fmf 
(tmt) huanli@kvm-01-guest01:foo$ tmt t ls
/a
(tmt) huanli@kvm-01-guest01:foo$ tmt t lint
/a
pass test script must be defined
pass directory path must be absolute
pass directory path must exist
warn summary is very useful for quick inspection
pass correct attributes are used

(tmt) huanli@kvm-01-guest01:foo$ echo $?
0
```
